swim/update_rollup_test.go

```diff
@@ -99,11 +100,23 @@ func (s *RollupTestSuite) TestFlushesAfterInterval() {
 	})
 
 	s.NotNil(s.r.FlushTimer(), "expected timer to be set")
+
+	// This is a race condition which never got triggered yet. If the scheduler
+	// decides to yield at this comment and proceed with flushing the buffer,
+	// assertion below will fail. Oops.
 	s.Len(s.r.Buffer(), 2, "expected two updates to be in buffer")
 
-	time.Sleep(2 * time.Millisecond)
+	// Flushing the buffer used to take <2ms on an otherwise idle machine. On
+	// travis-ci we might have a significantly less CPU shares, so wait longer.
+	sleepUpTo := 100 // Milliseconds
```

Comment:
This is a little misleading. 100 is the number if iterations of the loop, not the time. The call to `Buffer()` in the loop has a read lock, so I'd say it's entirely possible for the actual time spent to go past 100 milliseconds.